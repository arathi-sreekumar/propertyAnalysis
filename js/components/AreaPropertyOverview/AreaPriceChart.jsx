import React from 'react';
// import Chart from 'react-d3-core';
import { LineChart } from 'react-easy-chart';
import { arrayOf, shape, string, number } from 'prop-types';

class AreaPriceChart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: props.data
    };
  }

  render() {
    const width = 700;
    const height = 300;
    // const margins = { left: 100, right: 100, top: 50, bottom: 50 };
    const chartData = [this.state.data];

    // const title = 'Property price in the area over last 5 years';
    return (
      <div>
        <LineChart
          xType={'text'}
          axisLabels={{ x: 'Last 5 years', y: 'Average property price' }}
          axes
          dataPoints
          width={width}
          height={height}
          interpolate={'cardinal'}
          data={chartData}
        />
      </div>
    );
  }
}

AreaPriceChart.propTypes = {
  data: arrayOf(
    shape({
      x: string,
      y: number
    })
  ).isRequired
};

export default AreaPriceChart;
