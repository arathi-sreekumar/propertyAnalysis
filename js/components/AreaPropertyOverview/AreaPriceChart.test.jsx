import React from 'react';
import { shallow } from 'enzyme';
import AreaPriceChart from './AreaPriceChart';

const CHART_DATA = [
	{ x: 'present', y: 2100500 },
	{ x: '3 months', y: 2120500 },
	{ x: '6 months', y: 2150000 },
	{ x: '1 year', y: 2000200 }
];

describe('AreaPriceChart', () => {
	it('renders correctly', () => {
		const component = shallow(<AreaPriceChart data={CHART_DATA} />);
		expect(component).toMatchSnapshot();
	});
});
