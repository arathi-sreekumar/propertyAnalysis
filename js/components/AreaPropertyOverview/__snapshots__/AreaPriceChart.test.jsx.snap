// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaPriceChart renders correctly 1`] = `
<div>
  <LineChart
    axes={true}
    axisLabels={
      Object {
        "x": "Last 5 years",
        "y": "Average property price",
      }
    }
    clickHandler={[Function]}
    data={
      Array [
        Array [
          Object {
            "x": "present",
            "y": 2100500,
          },
          Object {
            "x": "3 months",
            "y": 2120500,
          },
          Object {
            "x": "6 months",
            "y": 2150000,
          },
          Object {
            "x": "1 year",
            "y": 2000200,
          },
        ],
      ]
    }
    dataPoints={true}
    datePattern="%d-%b-%y"
    height={300}
    interpolate="cardinal"
    lineColors={Array []}
    mouseMoveHandler={[Function]}
    mouseOutHandler={[Function]}
    mouseOverHandler={[Function]}
    width={700}
    xType="text"
    yType="linear"
  />
</div>
`;
